apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts

```diff
@@ -618,7 +658,15 @@ export class BuyPlanComponent implements OnInit, OnDestroy {
     public toggleDuration(event: any): void {
         if (event) {
             this.firstStepForm.get('duration').setValue(event?.value);
-            this.setPlans();
+            if (!this.subscriptionId) {
+                this.setPlans();
+            } else {
+                this.inputData = [];
+                const filteredPlans = this.firstStepForm.get('duration')?.value === 'YEARLY' ? this.yearlyPlans : this.monthlyPlans;
+                filteredPlans?.forEach(plan => {
```

Comment:
```suggestion
                this.inputData.push(...filteredPlans);
```

Reply:
You may also write like this without loop

Reply:
DONE